src/eva/core/callbacks/writers/embeddings.py

```diff
@@ -85,10 +85,12 @@ def write_on_batch_end(
             input_name, save_name = self._construct_save_name(
                 dataset.filename(global_idx), metadata, local_idx
             )
+            save_name = save_name.split(".")[0] + f"_{global_idx}.pt"
```

Comment:
Related to above comment, this change shouldn't be necessary if we use `group_key=slide_id` I believe